### PR TITLE
Use swiftclient.utils.generate_temp_url

### PR DESCRIPTION
**Pull request by einarf**
Suggestion for #79

### CODE REVIEW

File: swift/storage.py [hunk 1]
```diff
@@ -360,7 +361,8 @@ def _path(self, name):
             path = urlparse.urlsplit(url).path
             msg = ('%s\n%s\n%s' % (method, expires, path)).encode()
             sig = hmac.new(self.temp_url_key, msg, sha1).hexdigest()
-            url = url + '?temp_url_sig=%s&temp_url_expires=%s' % (sig, expires)
+            tmp_path = generate_temp_url(path, expires, sig, method, absolute=True)
```
Comment (dperetti) [inline swift/storage.py]: I think we should not pass `sig` here, but `self.temp_url_key`, [according to the docs](http://docs.openstack.org/developer/python-swiftclient/swiftclient.html#swiftclient.utils.generate_temp_url).

File: swift/storage.py [hunk 2]
```diff
@@ -357,11 +355,8 @@ def _path(self, name):
         # Are we building a temporary url?
         if self.use_temp_urls:
             expires = int(time() + int(self.temp_url_duration))
-            method = 'GET'
             path = urlparse.urlsplit(url).path
```
Comment (dperetti) [inline swift/storage.py]: Should be :

    path = urlparse.unquote(urlparse.urlsplit(url).path)

..because `name` was quoted above (line 355).

**Comment by einarf** [comment]
@blacktorn This one is more than ready now 😃

**Comment by einarf** [comment]
@blacktorn Once again ready for merge.

**Pull request merged**